lib/smart_answer_flows/maternity-paternity-calculator/adoption_calculator_flow.rb

```diff
@@ -222,24 +178,14 @@ def define
           from { 2.years.ago(Time.zone.today) }
           to { 2.years.since(Time.zone.today) }
 
-          precalculate :payday_offset do
-            calculator.payday_offset
-          end
-
-          precalculate :payday_offset_formatted do
-            calculator.format_date_day payday_offset
-          end
```

Comment:
Are these being used by the question template? Can't see where these values are being set?

Reply:
To be honest, I really struggled to find where some variables were used. A problem is that the system is so complicated that I had to fall back to relying on the tests, but the tests often test that a particular attribute is set as expected, without it being used. 

Due to the way flows behave, things are further complicated by the tests not testing an attribute in isolation. That is, to test C, the tests often test A and B first (particularly in "integration" tests), and often it is not clear whether A and/or B are prerequisites of C or are just set on the way. That can make it hard to cut an attribute test out.

As a result, I've tended to refactor to maintain the current behaviour and only "corrected" that behaviour if the change was simple and obvious.

Reply:
Yup - that makes sense and is the safest!